### PR TITLE
Orbit Panel Overhaul: Antag Icons, Collapsibility, Generic Theme

### LINKED ISSUE #8910: Orbit Menu Runtimes
## Occurance Details

**Round Date:** 2023-04-21

**Round ID:** 43693

**Runtime Information:**
```
[15:08:16] Runtime in job.dm, line 104: job name Personal AI is not valid
proc name: GetJob (/datum/controller/subsystem/job/proc/GetJob)
usr: #Redacted#/(#Redacted#)
usr.loc: (Robotics Lab (166, 119, 2))
src: Jobs (/datum/controller/subsystem/job)
call stack:
Jobs (/datum/controller/subsystem/job): GetJob("Personal AI")
/datum/orbit_menu (/datum/orbit_menu): ui data(#Redacted# (/mob/dead/observer))
/datum/tgui (/datum/tgui): get payload(null, 1, 1)
/datum/tgui (/datum/tgui): open()
/datum/orbit_menu (/datum/orbit_menu): ui interact(#Redacted# (/mob/dead/observer), /datum/tgui (/datum/tgui))
#Redacted# (/mob/dead/observer): Orbit()
Orbit (/atom/movable/screen/ghost/orbit): Click(null, "mapwindow.map", "icon-x=21;icon-y=10;left=1;scr...")
```

### PR DESCRIPTION
**Pull request by itsmeow**
## About The Pull Request

#8252 introduced a bunch of issues, this fixes them and also adds antag icons for ghost-visible antags.

- Job icons now have proper margins
- Line height is now consistent between job buttons with or without icons, making the UI flow properly.
- Antag icons will now show on ghost visible antags.
- Now uses generic UI theme, as it's not an IC element.
- Adds collapsibility to the menu sections
- Changes the primary color of the generic theme so section headers, etc are the correct color.
- Orbit menu now updates instead of opening new orbit menus.

Fixes #8910

## Why It's Good For The Game

Fixes bugs and makes the UI look better

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

visible runtimes are unrelated to this PR and are from my broken DB

image

</details>

## Changelog
:cl:
fix: Having an invalid job will no longer cause a runtime in the orbit menu.
fix: Orbit menu now spaces job icons properly and has consistent button height.
add: Orbit menu will now show antag icons for ghost-visible antagonists.
tweak: Orbit menu will now use generic UI theme
tweak: Generic UI theme now colors more elements (Section headers, input outlines, etc)
tweak: Orbit menu sections are now collapsible.
tweak: Orbit menu will now properly update instead of opening new orbit menus when clicked. It will also autoupdate with new data.
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/observer/orbit.dm [hunk 1]
```diff
@@ -65,7 +65,7 @@
 				var/obj/item/card/id/identification_card = M.get_idcard()
 				if (identification_card)
 					serialized["role_icon"] = "hud[ckey(identification_card.GetJobIcon())]"
-				else
+				else if(mind.assigned_role in SSjob.name_occupations)
```
Comment (EvilDragonfiend) [inline code/modules/mob/dead/observer/orbit.dm]:
```suggestion
				else if(SSjob.name_occupations[mind.assigned_role])
```

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]

File: code/modules/mob/dead/observer/orbit.dm (hunk 1 above)
Comment (itsmeow) [reply code/modules/mob/dead/observer/orbit.dm]: They do the same thing

Comment (EvilDragonfiend) [reply code/modules/mob/dead/observer/orbit.dm]: > They do the same thing

https://www.byond.com/docs/ref/#/operator/in

It does the same thing yeah, but it's related with time coplexity. `A in B` is actually the same as `for(var/each in B)` and it is O(n) which means can possibly worse if the target job is at the end of the list, while `B[A]` directly refers the job from the list which is O(1)

Comment (itsmeow) [reply code/modules/mob/dead/observer/orbit.dm]: That's dumb

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**